src/main/java/io/github/mzmine/parameters/parametertypes/ranges/MobilityRangeParameter.java

```diff
@@ -0,0 +1,60 @@
+package io.github.mzmine.parameters.parametertypes.ranges;
+
+import java.text.NumberFormat;
+import java.util.Collection;
+import com.google.common.collect.Range;
+
+public class MobilityRangeParameter extends DoubleRangeParameter {
+
+  public MobilityRangeParameter() {
+    super("Mobility", "Mobility depends on acqusition technique, e.g. drift tube in ms", null, true,
+        null);
+  }
+
+  public MobilityRangeParameter(boolean valueRequired) {
+    super("Mobility", "Mobility depends on acqusition technique, e.g. drift tube in ms", null,
+        valueRequired, null);
+  }
+
+  public MobilityRangeParameter(String name, String description, boolean valueRequired,
+      Range<Double> defaultValue) {
+    super(name, description, null, valueRequired, defaultValue);
+  }
+
+  public MobilityRangeParameter(String name, String description, NumberFormat format,
+      boolean valueRequired, Range<Double> defaultValue) {
+    super(name, description, format, valueRequired, defaultValue);
+  }
+
+  @Override
+  public boolean checkValue(Collection<String> errorMessages) {
+    if (valueRequired && (this.getValue() == null)) {
```

Comment:
If a scan selection is put for a dialog with no IMSRawDataFileSelected, the value will be null (currently) and thus block thevalue check here, or?

Reply:
no, I have added the MobilityParameter with valueRequired set to false